server/app/api.py

```diff
@@ -865,6 +868,11 @@ def autograde(self, obj, user, data):
       if 'grade_final' in data and data['grade_final']:
         #Collect all final submissions and run grades.
         deferred.defer(autograde_final_subs, obj, user, data)
+    
+        if 'promote_backups' in data and data['promote_backups']:
+            # Force promote backups and run autograder
+            defrred.defer(promote_student_backups, obj, True, user, data)
+
```

Comment:
Oh, promote_student_backups also runs the autograder on those backups. It may be cleaner to separate this into two steps: promotion and grading